apply_DDT.py

```diff
@@ -0,0 +1,574 @@
+    parser.add_argument('--sig-files', default='data/sig_20240718/sig_*/*.npz', help='Signal data files')
+
+    # BDT and ddt model
+    parser.add_argument('--bdt_file', default='models/svjbdt_obj_rev_version.json', help='BDT model file')
+    parser.add_argument('--ddt_map_file', default='models/ddt_obj_rev_version.json', help='DDT map file')
+
+    parser.add_argument('--lumi', type=float, default=137600, help='Luminosity')
+
+    parser.add_argument('--sig_bdt_cut', type=float, default=0.65, help='BDT cut for signal plotting')
+
+    # Choose the BDT cut values that you want to make for the DDT
+    # or that are in the DDT you are loading
+    # another common set of cuts is
+    # bdt_cuts = [0.0, 0.1, 0.2, 0.3, 0.4, 0.42, 0.45, 0.47, 0.5, 0.52, 0.55, 0.57, 0.6, 0.62, 0.65, 0.67, 0.7, 0.72, 0.75, 0.77, 0.8, 0.82, 0.85, 0.87, 0.9, 0.92, 0.95] 
+    # and another common set for plots is [0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9], help='List of BDT cuts')
+    parser.add_argument('--bdt_cuts', nargs='+', type=float, default=[0.0, 0.1, 0.2, 0.3, 0.4, 0.42, 0.45, 0.47, 0.5, 0.52, 0.55, 0.57, 0.6, 0.62, 0.65, 0.67, 0.7, 0.72, 0.75, 0.77, 0.8, 0.82, 0.85, 0.87, 0.9, 0.92, 0.95], help='List of BDT cuts')
+
+    # Select the plots to make
+    parser.add_argument('--plot_2D_DDT_map', action='store_true', help='Plot 2D DDT map')
+    parser.add_argument('--plot_bkg_scores_mt', action='store_true', help='Plot background scores mt')
+    parser.add_argument('--plot_fom_significance', action='store_true', help='Plot FOM significance')
+    parser.add_argument('--plot_sig_mt_single_BDT', action='store_true', help='Plot signal mt single BDT')
+    parser.add_argument('--plot_one_sig_mt_many_bdt', action='store_true', help='Plot one signal mt many BDT')
+
+    return parser.parse_args()
+
+#------------------------------------------------------------------------------
+# User defined functions ------------------------------------------------------
+#------------------------------------------------------------------------------
+
+def FOM(s, b):
+    '''
+    The significance for signal vs. background derived from the CLs method
+    '''
+    FOM = np.sqrt(2*((s+b)*np.log(1+s/b)-s))
+    return FOM
+
+def bdt_ddt_inputs(col, lumi, all_features):
+
+    # Storing column features
+    X = []
+    weight = []
+    
+    # Test if more than one column
+    if isinstance(col,list):
+        for icol in col:  
+            # Grab the input features and weights
+            X.append(icol.to_numpy(all_features))
+            weight.append(icol.xs / icol.cutflow['raw'] * lumi * icol.arrays['puweight'])
+        X = np.concatenate(X)
+        weight = np.concatenate(weight)
+
+    # If only one column
+    else: 
+        X = col.to_numpy(all_features)
+        weight = col.xs / col.cutflow['raw'] * lumi * col.arrays['puweight']
+  
+    # Apply the signal region
+    rt = X[:,-1]
+    X = X[rt > 1.18]
```

Comment:
please use `common.apply_rt_signalregion()` to ensure consistency.

Reply:
Added, needs testing